front/lib/workspace_usage.ts

```diff
@@ -473,9 +474,9 @@ export async function checkWorkspaceActivity(workspace: Workspace) {
   const sevenDaysAgo = new Date();
   sevenDaysAgo.setDate(sevenDaysAgo.getDate() - 7);
 
-  const hasDataSource = await DataSource.findOne({
-    where: { workspaceId: workspace.id },
-  });
+  const hasDataSource = await DataSourceResource.workspaceHasDatasources(
```

Comment:
This should probably live at the workspace level, instead can we do a regular `DataSourceResource.listXX()` for now and check that it returns more than 1 row? Maybe add a limit to this function.